accepted/00000-modular-repos-with-clm.md

```diff
@@ -0,0 +1,194 @@
+
+It utilizes the Python port of [libmodulemd](https://github.com/fedora-modularity/libmodulemd), which is used for structural parsing of `modules.yaml` files.
+An example of such implementation is provided in [attachments/00000-modular-repos-api.py](attachments/00000-modular-repos-api.py).
+
+### UI
+The UI consists of a new type of CLM filter called "module". The filter provides inputs for a module name and a stream name.
+Each filter selects a stream for a single module. The stream input can be left empty to select the default stream.
+Since a CLM filter is not related to any channels, it is not possible to display a list of modules/streams on the filter form and the inputs can only be simple text inputs.
+The module information for a modular repository is displayed under an additional "Modules" tab on the "Channel Details" page.
+A modular repository is indicated with an icon next to the name of the repository in the CLM sources list. A click on the icon navigates to the "Modules" tab of the channel details.
+
+## The Process
+
+This section describes the detailed design for the different phases of a CLM project.
+
+### Project setup
+At the project setup, the backend checks if any of the sources are modular repositories.
+This check is done by doing a `getChannel().getModule()` call on the source object.
+For every modular source, a proper indicator is displayed as described in the UI component section.
+
+Currently, every CLM filter displays a field to choose one of two hard-coded rules: `allow` or `deny`. Since the module filtering logic works in an implicit "deny all" fashion (See build/promote section), this field must be hidden for module filters.
+Each added module filter stores the value of a chosen module/stream in `name` or `name:stream` format.
+
+Whenever a change is made in any of the filters, a call to the same API endpoint is made to check the validity of the current selections to provide real-time feedback when the current combination is not a valid one. The API's error reporting mechanism is used to determine the failure and provide meaningful messages to the user in the UI.
+
+### Build/promote
+At build time, all the module filter values are read in the form of `name:stream` tuples or only `name` in case the default stream is chosen.
+The remaining modules where a selection has not been made are set to disabled (except the case that no module filters are attached at all, see below).
+With a list of name/stream tuples, the API is called to resolve the modular dependencies and the appropriate contexts (see next section).
+If successful, the API returns a list of full RPM names belonging to all the enabled streams.
+Additionally, the backend retrieves a full list of all the modular RPMs and subtracts the enabled ones from this list. The resulting list is translated into corresponding `Package` objects.
```

Comment:
We need to check what happens to `epoch` in the full package names. In SUSE Manager the epoch is not part of the filename which create some trouble. We should check the artifact list if they contain the epoch or not.
If they do we are fine. If not, we can expect some extra work.